dts/bindings/rtc/nxp,irtc.yaml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: IRTC
+
+compatible: "nxp,irtc"
+
+include: rtc.yaml
+
+properties:
+  reg:
+    required: true
+
+  clock-select:
+    type: int
+    required: true
+    description: |
+      The input clock select for IRTC.
+      0 <- 16.384 kHz
+      1 <- 32.768 kHz
+
+  alarms-count:
+    type: int
+    required: true
+    description: Number of alarms supported by IRTC.
+
+  output-clk-en:
+    type: boolean
+    description: |
+      The IRTC can send its clock as an output source
+      for other peripherals to use.
+      "" <- Output Clock Enabled
+      "enable-output-clock" <- Enable Output Clock
```

Comment:
these two lines don't make any sense to me

Reply:
Typo, one should be Enable and one should be Disabled now, but I was just demonstrating how this is used in devicetree to enable or disable this value.

Reply:
looking at this application node https://www.nxp.com/docs/en/application-note/AN4023.pdf:

* 0—No clock is output
* 1—Compensated 1 Hz
* 2—Buffered oscillator clock

maybe change this to an int?
```
  output-clk:
    type: int
    default: 0
    description: |
      0 -> No clock output
      1 -> Compensated 1Hz
      2 -> Buffered oscillator clock
    enum:
      - 0
      - 1
      - 2
```
not sure if this specific chip supports it, but seemingly some of them do :)

Reply:
Found this:
```
/*! CLKOUT - RTC Clock Output Selection
 *  0b00..No output clock
 *  0b01..Fine 1 Hz clock with both precise edges
 *  0b10..32.768 or 16.384 kHz clock
 *  0b11..Coarse 1 Hz clock with both precise edges
 */
#define RTC_CTRL_CLKOUT(x)                       (((uint16_t)(((uint16_t)(x)) << RTC_CTRL_CLKOUT_SHIFT)) & RTC_CTRL_CLKOUT_MASK)
```

Reply:
Will change the Description on the next update.